includes/abstracts/abstract-wc-data.php

```diff
@@ -479,8 +477,12 @@ public function save_meta_data() {
 			} elseif ( empty( $meta->id ) ) {
 				$new_meta_id                       = $this->data_store->add_meta( $this, $meta );
 				$this->meta_data[ $array_key ]->id = $new_meta_id;
```

Comment:
This `$meta === $this->meta_data[ $array_key ]` is TRUE (objects are always by reference unless `clone $object` is used). But that's a nice catch anyway, I will make that line shorter. Thanks!